cmd/x.go

```diff
@@ -0,0 +1,67 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/exoscale/egoscale"
+
+	"github.com/exoscale/cli/cmd/internal/x"
+)
+
+var xCmd *cobra.Command
+
+func init() {
+	xCmd = x.InitCommand()
+	xCmd.Use = "x"
+	xCmd.Hidden = true
+	xCmd.Long = `Low-level Exoscale API calls -- don't use this unless you have to.
+
+These commands are automatically generated using openapi-cli-generator[1],
+input parameters can be supplied either via stdin or using Shorthands[2].
+
+[1]: https://github.com/danielgtaylor/openapi-cli-generator
+[2]: https://github.com/danielgtaylor/openapi-cli-generator/tree/master/shorthand
+`
+
+	// This code being executed very early, at this point some information is not ready
+	// to be used such as the account's configuration and some global variables, so we
+	// we use the subcommand pre-run hook to perform last second changes to the
+	// outgoing requests.
+	xCmd.PersistentPreRunE = func(cmd *cobra.Command, _ []string) error {
+		// If no value is provided for flag `--server`, infer the server URL from the current
+		// profile's default zone or explicit `--zone` flag if specified:
+		if server, _ := cmd.Flags().GetString("server"); server == "" {
+			zone := gCurrentAccount.DefaultZone
+			if z, _ := cmd.Flags().GetString("zone"); z != "" {
+				zone = z
+			}
+
+			env, err := cmd.Flags().GetString("environment")
```

Comment:
I still think we should allow setting the environment in our profiles (the following would assume there's an `Env` field in accounts:

```suggestion
			env, _ := cmd.Flags().GetString("environment"); env == "" {
			  if (gCurrentAccount.Env != "") {
			      env = gCurrentAccount.Env
			  } else {
			    env = "api"
			  }
			}
```

Reply:
Done.